### DISCUSSION TITLE
SPI_FREQUENCY -- is this proper use or PEBCAK?

### DISCUSSION THREAD

**Discussion by chconnor** [Bodmer/TFT_eSPI#2954] [Q&A - General] [answered]
Hi -- vscode + PlatformIO driving an ILI9488 on an ESP32-DevKitC-VIE, TFT_eSPI v2.5.33

Display works fine, I was just trying to get to the bottom of some SPI frequency questions I had.

I use a copied/customized version of USER_SETUP_ID 21 which #define's the SPI_FREQUENCY as 27MHz.

In my code I start the screen and later print the frequency of the SPI bus with:

```Serial.println(spiClockDivToFrequency(spiGetClockDiv(SPI.bus())))```

...but I always get 80MHz.

If I do not use the screen at all, and instead I initialize an SD card reader on the same SPI bus with it's default setting of 16MHz, the same code reports 16MHz, as expected.

This implies to me that maybe TFT_eSPI isn't setting the bus frequency to the requested value, but I'm too much of a beginner to understand if this is accurate. Am I misunderstanding something fundamental about SPI, the associated functions, and TFT_eSPI? Thanks!

**Comment by Bodmer** [comment, marked as answer]
The library creates a new SPI instance which will have different settings to the default SPI instance.

There is a [function](https://github.com/Bodmer/TFT_eSPI/blob/master/TFT_eSPI.h#L824) to get the SPI instance created by the library.

**Reply by chconnor** [reply]
PEBCAK it was. :-) Thanks for the response.
